### PR TITLE
Allow underscores in Concept ID (Issue #16)

### PR DESCRIPTION
**Pull request by davetrig**
Made changes to Concept model and serializer to permit underscores in mnemonic field (Concept ID)
Changed hard-coded regex patterns in various urls.py files to use constants

### CODE REVIEW

File: django-nonrel/ocl/concepts/models.py [hunk 1]
```diff
@@ -50,7 +50,8 @@ def is_search_index_term(self):
 CONCEPT_TYPE = 'Concept'
 
 
-class Concept(ConceptValidationMixin, SubResourceBaseModel, DictionaryItemMixin):
+class Concept(ConceptValidationMixin, ConceptBaseModel, DictionaryItemMixin):
```
Comment (rkorytkowski) [inline django-nonrel/ocl/concepts/models.py]: Why we need to change to ConceptBaseModel here?

Comment (rkorytkowski) [inline django-nonrel/ocl/concepts/models.py]: Nevermind, saw the explanation below.

**Comment by rkorytkowski** [comment]
The code looks good, but we need to have some tests. Could you please implement at least one test for each resource?

**Comment by davetrig** [comment]
@rkorytkowski, I've created a test "test_create_concept_positive__mnemonic_with_underscore", similar to "test_create_concept_positive" in concepts.tests.ConceptTest, but being unfamiliar with the test system, I'm unclear as to what else is required here.  I've committed this change to my fork; can you take a look and let me know what else is needed to fulfill the requirement?

**Comment by rkorytkowski** [comment]
@davetrig I think you forgot to push changes to your fork... I don't see test_create_concept_positive__mnemonic_with_underscore

**Comment by davetrig** [comment]
You're correct; I forgot to push the commit.  It's there now.

**Pull request merged**